docs/blog/posts/chain-of-density.md

```diff
@@ -0,0 +1,467 @@
+    of the benefits that `Instructor` provides. With a simple `patch`, we can get
+    **automatic type coercion of our outputs and automatic retries for invalid outputs**
+    out of the box!
+
+2.  We first generate an initial summary. Note here that we explictly ask for a summary that has
+    80 words and is lengthy with overly verbose fillers in the system prompt
+
+3.  We slightly modify the original system prompt used in the original paper to perform a rewrite of the summary.
+    Using `Instructor`, we also get validation of the generated output with our `field_validator`s that we defined above
+
+4.  If you've chosen a value that is larger than 0.08, make sure to increase this value in case you need to do multiple rewrites
+
+This summarization function yields a result which triples the number of entities while mantaining the same number of tokens. We can also see that stylistically, the summary is a lot more natural.
+
+**First Iteration**
+
+> This article discusses the highly-anticipated boxing match between Manny Pacquiao and Floyd Mayweather. The article revolves around Manny Pacquiao's statements about his upcoming fight and his preparations for the same. A portion of the article provides details about the financial stipulations of the match and its significance in the sporting arena. Quotes from Pacquiao illustrating his determination and his battle strategy are highlighted. The tone of the article is largely centered around creating a build-up to the upcoming mega event.
+
+**Final Iteration**
+
+> Manny Pacquiao, the Filipino boxer, anticipates the forthcoming May 2 showdown at the MGM Grand as the fight of his life, against the undefeated American Floyd Mayweather, in a $300m bout. Despite being seen as the underdog in this high-stakes Las Vegas match, Pacquiao is confident, promising a warrior's spirit and assuring the fans who have been awaiting this encounter for a decade, that it will indeed be the biggest sporting spectacle in history worthy of their anticipation
+
+## Part 2) Fine-Tuning
+
+In this section, we'll look into how to fine-tune a GPT 3.5 model so that it is able to perform at an equivalent level as a GPT-4 model. We'll then compare the performance of our model against that of `GPT-4` and `GPT-4-Turbo` to see how it stacks up.
+
+### Creating a Training Set
+
+Let's first segregate our train and test set so that we don't have any sort of contamination - this corresponds to our `train.csv` and `test.csv` in our [Hugging Face Dataset](https://huggingface.co/datasets/ivanleomk/gpt4-chain-of-density). Now, we just need to import the `Instructions` module from the `Instructor` package which allows you to generate a nicely formatted `.jsonl` file to be used for fine-tuning
+
+```py hl_lines="2 9 11-18 37 40"
+from typing import List
+from chain_of_density import summarize_article #(1)!
+import csv
+import logging
+import instructor
+from pydantic import BaseModel
+
+logging.basicConfig(level=logging.INFO) #(2)!
```

Comment:
This feels like an unnecessary detail that the user has to be careful not to trip over—perhaps `instructor.Instructions` can have it's own logging handler so the user doesn't have to even be aware of it?

Reply:
@jxnl thoughts?